docs/network_architecture.rst

```diff
@@ -376,14 +372,13 @@ this conversion.
 Training Round Communication
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The communication is
-summarised in the following sequence diagram.
+The communication is summarised in the following sequence diagram.
 In a training round, :math:`C` is in the state :code:`ROUND`. The selected
 participant :math:`P` is in :code:`TRAINING` state. The first message is by
 :math:`P` essentially kicks off the exchange. :math:`C` responds with the global
-model :math:`\theta` (and other data as specified in
-:code:`StartTrainingReply`). Then :math:`P` carries out the training locally.
-When complete, it sends the updated model :math:`\theta'` (and other metadata)
+model :math:`\weights` (and other data as specified in
+:code:`StartTrainingRoundResponse`). Then :math:`P` carries out the training locally.
+When complete, it sends the updated model :math:`\weights'` (and other metadata)
 back. :math:`C` responds with an acknowledgement.
 
 .. image:: _static/sequence2.png
```

Comment:
ideally the sequence diagram should be updated with the new names too. It's unfortunate that the Sphinx markup doesn't support this kind of diagram (or it does IIRC but looks quite ugly in comparison)... I can help out with updating the image if you want.

Reply:
I agree with that but can we not block merging the PR because of this? Merging it as is would already fix the tests, and we could update the doc in a follow-up PR

Reply:
yes i would not consider this a blocker, up to @janpetschexain whether to resolve this here or later!

Reply:
replaced the diagram and added the source code as a comment.